flytekit/common/tasks/task.py

```diff
@@ -91,6 +91,10 @@ def promote_from_model(cls, base_model):
             custom=base_model.custom,
             container=base_model.container
         )
+        # Override the newly generated name if one exists in the base model
+        if base_model.id:
```

Comment:
does this check work? I imagine it will always be true.

Reply:
feel more comfortable with this, leaving for now unless you really want me to remove it.

Reply:
The check is fine, i just don't think it will work even if the ID is "empty"